src/hash.cr

```diff
@@ -272,6 +272,26 @@ class Hash(K, V)
     self
   end
 
+  def select(&block : K, V -> U)
+    inject({} of K => V) do |memo, k, v|
+      yield(k, v) ? memo.merge!({k => v}) : memo
```

Comment:
Our Enumerable has it already in fact. Atm we can have Hash include Enumerable because there's no autosplatting and thus its `#each` becomes incompatible with what Enumerable demands.